jax/_src/random.py

```diff
@@ -2042,18 +2042,22 @@ def orthogonal(
   key: ArrayLike,
   n: int,
   shape: Shape = (),
-  dtype: DTypeLikeFloat = float
+  dtype: DTypeLikeFloat = float,
+  m: int | None = None,
 ) -> Array:
   """Sample uniformly from the orthogonal group O(n).
 
   If the dtype is complex, sample uniformly from the unitary group U(n).
 
+  For unequal rows and columns, this samples a semi-orthogonal matrix instead.
```

Comment:
Maybe say more here, since I'm not sure that "semi-orthogonal" is widely understood. We should also mention (if I'm understanding the intent correctly) that if `n < m`, then the rows will be mutually orthogoal, and for `n > m` then the columns will be mutually orthogonal.